tests/platform_helper/utils/test_validation.py

```diff
@@ -419,6 +423,208 @@ def test_validate_s3_bucket_name_multiple_failures():
 
 
 @patch("dbt_platform_helper.utils.validation.warn_on_s3_bucket_name_availability", new=Mock())
-def test_validate_success(valid_platform_config):
-    PLATFORM_CONFIG_SCHEMA.validate(valid_platform_config)
+@patch("dbt_platform_helper.utils.validation.get_aws_session_or_abort", new=Mock())
+def test_validate_platform_config_success(valid_platform_config):
+    validate_platform_config(valid_platform_config)
     # No assertions - validate will error if config is invalid.
+
+
+@pytest.mark.parametrize(
+    "account, envs, exp_bad_envs",
+    [
+        ("non-prod", ["dev"], ["dev"]),
+        ("non-prod-acc", ["prod"], ["prod"]),
+        ("prod-acc", ["dev", "prod"], ["dev"]),
+        ("prod-acc", ["dev", "staging", "prod"], ["dev", "staging"]),
+        ("non-prod-acc", ["dev", "prod"], ["prod"]),
+    ],
+)
+@patch("dbt_platform_helper.utils.validation.warn_on_s3_bucket_name_availability", new=Mock())
+@patch("dbt_platform_helper.utils.validation.abort_with_error")
+@patch("dbt_platform_helper.utils.validation.get_aws_session_or_abort", new=Mock())
+def test_validate_platform_config_fails_if_pipeline_account_does_not_match_environment_accounts(
+    mock_abort_with_error, platform_env_config, account, envs, exp_bad_envs
+):
+    platform_env_config["environment_pipelines"] = {
+        "main": {
+            "account": account,
+            "slack_channel": "/codebuild/notification_channel",
+            "trigger_on_push": True,
+            "environments": {env: {} for env in envs},
+        }
+    }
+
+    validate_platform_config(platform_env_config)
+
+    message = mock_abort_with_error.call_args.args[0]
+
+    assert "The following pipelines are misconfigured:" in message
+    assert (
+        f"  'main' - these environments are not in the '{account}' account: {', '.join(exp_bad_envs)}"
+        in message
+    )
+
+
+@patch("dbt_platform_helper.utils.validation.warn_on_s3_bucket_name_availability", new=Mock())
+@patch("dbt_platform_helper.utils.validation.abort_with_error")
+@patch("dbt_platform_helper.utils.validation.get_aws_session_or_abort", new=Mock())
+def test_validate_platform_config_fails_if_pipeline_account_does_not_match_environment_accounts_multiple_pipelines(
```

Comment:
Isn't this covered by the test above?

Reply:
There's only a single pipeline in the tests above. That exercises multiple environment combinations in a single pipeline. This tests a multi-pipeline setup. Could merge them as one test but the mark.parametrize would get gnarly.

Reply:
Perhaps we can rename the tests to express the intent more clearly?